smapply/api.py

```diff
@@ -232,13 +232,17 @@ def process_user(sma_user):
     Returns:
         User: user modified or created by the function
     """
+    serializer = UserSerializer(data=sma_user)
+    serializer.is_valid(raise_exception=True)
+
     user, _ = User.objects.get_or_create(
         email__iexact=sma_user['email'],
         defaults={'email': sma_user['email'], 'username': sma_user['email']}
     )
     profile, _ = Profile.objects.get_or_create(user=user)
     if not profile.smapply_id:
         profile.smapply_id = sma_user['id']
+        profile.smapply_user_data = sma_user
```

Comment:
Can you add a line in `test_process_new_profile` to check that this field is populated correctly?
```python
assert Profile.objects.get(smapply_id=sma_user['id']).smapply_user_data == sma_user
```